### PR TITLE
Add support for ref fields and scoped ref parameters to SyntaxGenerator

### LINKED ISSUE #68959: SyntaxGenerator ignores "scoped" keyword
**Version Used**: 
4.6.0

**Steps to Reproduce**:

1. Run attached sample.  See also https://sharplab.io/#v2:EYLgtghglgdgPgWAFAAEDMACFBGAbFgJgwGFkBvZDKrTHfHAOgGUAHCGAHgBUA+EgJwCmEAC6DW7bjwAUAZwDGAexaCAJhiEAzDFw2DNgoTHmCANBlgiMAG0EwA5iIAWASgxkMz/ooDuGGACu1tYA3BgAvsjhQA=

[syntaxGenTest.zip](https://github.com/dotnet/roslyn/files/12004694/syntaxGenTest.zip)

**Expected Behavior**:
Method signature emitted with `scoped` keyword.

**Actual Behavior**:
`scoped` keyword is omitted.

### PR DESCRIPTION
**Pull request by ericstj**
Fixes #68959

@CyrusNajmabadi

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 1]
```diff
@@ -191,9 +191,12 @@ public override SyntaxNode NamespaceDeclaration(SyntaxNode name, IEnumerable<Syn
         }
 
         private protected override SyntaxNode ParameterDeclaration(
-            string name, SyntaxNode? type, SyntaxNode? initializer, RefKind refKind, bool isExtension, bool isParams)
+            string name, SyntaxNode? type, SyntaxNode? initializer, RefKind refKind, bool isExtension, bool isParams, bool isScoped)
         {
-            var modifiers = CSharpSyntaxGeneratorInternal.GetParameterModifiers(refKind);
+            var modifiers = CSharpSyntaxGeneratorInternal.GetParameterModifiers(refKind);            
+            if (isScoped)
+              modifiers = modifiers.Insert(0, SyntaxFactory.Token(SyntaxKind.ScopedKeyword));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:198]: incorrect indentation.

File: src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs [hunk 2]
```diff
@@ -85,6 +85,7 @@ IMethodSymbol or
                     isSealed: symbol.IsSealed,
                     isConst: isConst,
                     isUnsafe: symbol.RequiresUnsafeModifier(),
+                    isRef: field?.RefKind == RefKind.Ref || field?.RefKind == RefKind.RefReadOnly,
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs:88]:
```suggestion
                    isRef: field?.RefKind is RefKind.Ref or RefKind.RefReadOnly,
```

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Comment by ericstj** [comment]
Looks like I can try `-runanalyzers -warnaserror /p:RoslynEnforceCodeStyle=true`

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**